docs/drafts/version-ranges.md

```diff
@@ -0,0 +1,93 @@
+# Operator version ranges
```

Comment:
I created a follow up issue to add example CRs to the doc: https://github.com/operator-framework/operator-controller/issues/552